modules/private.php

```diff
@@ -0,0 +1,298 @@
+		<?php
+	}
+
+	/**
+	 * Don't let search engines index private sites
+	 * or sites not deemed publicly available, like deleted, archived, spam.
+	 *
+	 * @param string $output Robots.txt output.
+	 */
+	static function private_robots_txt( $output ) {
+		if ( Jetpack_Private::is_private_blog() ) {
+			$output  = "User-agent: *\n"; // Purposefully overriding current output; we only want these rules.
+			$output .= "Disallow: /\n";
+		}
+
+		return $output;
+	}
+
+	/**
+	 * Filters the link title attribute for the message displayed in the 'At a Glance' dashboard widget.
+	 *
+	 * @param string $text Default attribute text.
+	 */
+	static function privatize_privacy_on_link_title( $text ) {
+		if ( Jetpack_Private::is_private_blog() ) {
+			return __( 'Your site is visible only to registered members', 'jetpack' );
+		}
+
+		return $text;
+	}
+
+	/**
+	 * Filters the link label for the message displayed in the 'At a Glance' dashboard widget.
+	 *
+	 * @param string $text Default text.
+	 */
+	static function privatize_privacy_on_link_text( $text ) {
+		if ( Jetpack_Private::is_private_blog() ) {
+			return __( 'Private', 'jetpack' );
+		}
+
+		return $text;
+	}
+
+	/**
+	 * Output the meta tag that tells Pinterest not to allow users to pin
+	 * content from this page.
+	 * https://support.pinterest.com/entries/21063792-what-if-i-don-t-want-images-from-my-site-to-be-pinned
+	 */
+	static function private_no_pinning() {
+		echo '<meta name="pinterest" content="nopin" />';
+	}
+
+	/**
+	 * Prevents ajax requests on private blogs for users who don't have permissions
+	 *
+	 * @param string    $action The Ajax nonce action.
+	 * @param false|int $result The result of the nonce check.
+	 */
+	static function private_blog_ajax_nonce_check( $action, $result ) {
+		global $current_user, $wpdb;
+
+		if ( is_super_admin() || ! Jetpack_Private::is_private_blog() ) {
+			return;
+		}
+
+		if ( 1 !== $result && 2 !== $result ) {
+			return;
+		}
+
+		if ( 'find-posts' !== $action && 'internal-linking' !== $action ) {
+			return;
+		}
+
+		// Make sure we are in the right code path, if not bail now.
+		if ( ! is_admin() || ( ! defined( 'DOING_AJAX' ) || ! DOING_AJAX ) ) {
+			return;
+		}
+
+		if ( ! Jetpack_Private::is_private_blog_user( $wpdb->blogid, $current_user ) ) {
+			wp_die( -1 );
+		}
+	}
+
+	/**
+	 * Disables WordPress Rest API for external requests
+	 */
+	static function disable_rest_api() {
+		global $pagenow, $current_user, $wpdb;
+
+		if ( ! Jetpack_Private::is_private_blog() ) {
+			return;
+		}
+
+		if ( $current_user && ( is_super_admin() || Jetpack_Private::is_private_blog_user( $wpdb->blogid, $current_user ) ) ) {
+			return;
+		}
+
+		die( __( 'This site is private.', 'jetpack' ) );
+	}
+
+	static function module_override( $modules ) {
+		$disabled_modules = array(
+			'publicize',
+			'sharedaddy',
+			'subscriptions',
```

Comment:
It may be good to add a few more modules to that list:
- `json-api` 
- `enhanced-distribution`
- `google-analytics`
- `photon`
- `sitemaps`
- `verification-tools`
- `wordads`

All those modules would have third-parties trying to hit the site, when that is not possible.

Reply:
Thanks, added 👍